src/analysis/normed_space/basic.lean

```diff
@@ -676,6 +676,35 @@ begin
   { exact smul_closed_ball' hc x r }
 end
 
+/-- A (semi) normed real vector space is homeomorphic to the unit ball in the same space.
+This homeomorphism sends `x : E` to `(1 + ∥x∥)⁻¹ • x`. -/
+def homeomorph_unit_ball {E : Type*} [semi_normed_group E] [semi_normed_space ℝ E] :
+  E ≃ₜ ball (0 : E) 1 :=
+{ to_fun := λ x,
+    begin
+      refine ⟨(1 + ∥x∥)⁻¹ • x, mem_ball_zero_iff.2 _⟩,
+      have : ∥x∥ < |1 + ∥x∥| := (lt_one_add _).trans_le (le_abs_self _),
+      rwa [norm_smul, real.norm_eq_abs, abs_inv, ← div_eq_inv_mul,
+        div_lt_one ((norm_nonneg x).trans_lt this)],
+    end,
```

Comment:
I'd be tempted to write this
```suggestion
{ to_fun := λ x, ⟨(1 + ∥x∥)⁻¹ • x, begin
    have : ∥x∥ < |1 + ∥x∥| := (lt_one_add _).trans_le (le_abs_self _),
    rwa [← mem_ball_zero_iff, norm_smul, real.norm_eq_abs, abs_inv, ← div_eq_inv_mul,
      div_lt_one ((norm_nonneg x).trans_lt this)],
  end⟩,
```
to make it clear the the data doesn't need tactic mode, only the proof.